tests/integration/traits/trait-parameter.spec.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  DevnetNetworkOrchestrator,
+  StacksTransactionMetadata,
+  stacksNodeVersion,
+} from "@hirosystems/stacks-devnet-js";
+import { StacksNetwork, StacksTestnet } from "@stacks/network";
+import {
+  AnchorMode,
+  PostConditionMode,
+  SignedContractCallOptions,
+  broadcastTransaction,
+  callReadOnlyFunction,
+  contractPrincipalCV,
+  cvToString,
+  makeContractCall,
+  makeContractDeploy,
+  responseErrorCV,
+  responseOkCV,
+  stringAsciiCV,
+  stringCV,
+  uintCV,
+} from "@stacks/transactions";
+import { Accounts } from "../constants";
+import {
+  Account,
+  DEFAULT_EPOCH_TIMELINE,
+  asyncExpectStacksTransactionSuccess,
+  buildDevnetNetworkOrchestrator,
+  deployContract,
+  getNetworkIdFromEnv,
+  waitForStacksTransaction,
+} from "../helpers";
+import { errorToCV } from "../pox/helpers";
+
+const fee = 2000;
+
+describe("trait parameter", () => {
+  let orchestrator: DevnetNetworkOrchestrator;
+  let version: string;
+  if (typeof stacksNodeVersion === "function") {
+    version = stacksNodeVersion();
+  } else {
+    version = "2.1";
+  }
+  const timeline = {
+    ...DEFAULT_EPOCH_TIMELINE,
+    epoch_2_2: 118,
+    pox_2_unlock_height: 119,
+  };
+
+  const contractAddress = Accounts.DEPLOYER.stxAddress;
+
+  const codeBodyTestTrait = `(define-trait test-trait
+  ((foo (uint) (response uint uint))))
+
+(define-read-only (foo-arg (f <test-trait>))
+  (contract-of f)
+)
+
+(define-public (call-foo (f <test-trait>) (a uint))
+  (contract-call? f foo a)
+)`;
+
+  const codeBodyImplTrait = `(define-public (foo (a uint))
+(ok a)
+)`;
+
+  const callReadOnlyTestTraitFooArg = (network: StacksNetwork) => {
+    return callReadOnlyFunction({
+      contractName: "test-trait",
+      contractAddress,
+      functionName: "foo-arg",
+      functionArgs: [contractPrincipalCV(contractAddress, "impl-trait")],
+      network,
+      senderAddress: Accounts.WALLET_1.stxAddress,
+    });
+  };
+
+  const callReadOnlyTestTraitCallFo = (
```

Comment:
```suggestion
  const callReadOnlyTestTraitCallFoo = (
```
(and the uses below)